src/platforms/platform.dom.js

```diff
@@ -268,12 +268,21 @@ function unwatchForRender(node) {
 }
 
 function addResizeListener(node, listener, chart) {
-	var expando = node[EXPANDO_KEY] || (node[EXPANDO_KEY] = {});
+	var expando = node[EXPANDO_KEY] = node[EXPANDO_KEY] || {};
 
 	// Let's keep track of this added resizer and thus avoid DOM query when removing it.
 	var resizer = expando.resizer = createResizer(throttled(function() {
 		if (expando.resizer) {
-			return listener(createEvent('resize', chart));
+			var aspectRatio = chart.options.maintainAspectRatio && chart.aspectRatio || null;
+			var container = aspectRatio && node.parentNode;
+			var w = container ? container.clientWidth : 0;
+			var ret = listener(createEvent('resize', chart));
+			if (container && container.clientWidth !== w && chart.canvas) {
```

Comment:
I don't think we need to test `chart.canvas`

Reply:
Some tests `destroy` chart after the first resize event (`canvas = null` there). So those throw without this test. Originally I placed this check in `resize`, but moved it here instead.